File: documentation/overview/webhooks/sample.cs
// Sample in c#

using System;
using System.Security.Cryptography;
using System.Text;

public class AES128CBCDecryptor
{
    public static string DecryptAES128CBC(string encryptedData, byte[] key)
    {
        if (key.Length != 16)
        {
            throw new ArgumentException("Key must be 16 bytes (AES-128 requires a 128-bit key).");
        }

        // Extract IV from the encrypted data (first 16 bytes)
        byte[] iv = Encoding.UTF8.GetBytes(encryptedData.Substring(0, 16));
        string base64Data = encryptedData.Substring(16);
        byte[] encryptedBytes = Convert.FromBase64String(base64Data);

        if (encryptedBytes.Length % 16 != 0)
        {
            throw new ArgumentException("Encrypted data length must be a multiple of the block size (16 bytes for AES).");
        }

        using (Aes aes = Aes.Create())
        {
            aes.Key = key;
            aes.IV = iv;
            aes.Mode = CipherMode.CBC;
            aes.Padding = PaddingMode.None;

            using (ICryptoTransform decryptor = aes.CreateDecryptor())
            {
                byte[] decryptedBytes = decryptor.TransformFinalBlock(encryptedBytes, 0, encryptedBytes.Length);

                // Remove PKCS7 padding
                int paddingLength = decryptedBytes[^1];
                if (paddingLength > 16)
                {
                    throw new ArgumentException("Invalid padding detected.");
                }

                byte[] unpaddedBytes = new byte[decryptedBytes.Length - paddingLength];
                Array.Copy(decryptedBytes, unpaddedBytes, decryptedBytes.Length - paddingLength);

                return Encoding.UTF8.GetString(unpaddedBytes);
            }
        }
    }

    public static void Main(string[] args)
    {
        string encryptedData = "<encrypted_string>";
        byte[] key = Encoding.UTF8.GetBytes("<webhook_key>");

        try
        {
            string decryptedText = DecryptAES128CBC(encryptedData, key);
            Console.WriteLine("Decrypted text: " + decryptedText);
        }
        catch (Exception ex)
        {
            Console.WriteLine("Error during decryption: " + ex.Message);
        }
    }
}

File: documentation/overview/webhooks/sample.py
## Package needed: pip install cryptography


from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.backends import default_backend
import base64

def decrypt_aes_128_cbc(encrypted_data: str, key: bytes) -> str:
    iv = encrypted_data[:16].encode('utf-8')

    data = encrypted_data[16:]

    encrypted_bytes = base64.b64decode(data)

    if len(key) != 16:
        raise ValueError("Key must be 16 bytes (AES-128 requires a 128-bit key).")

    if len(encrypted_bytes) % 16 != 0:
        raise ValueError("Encrypted data length must be a multiple of the block size (16 bytes for AES).")

    # Create the AES-128-CBC cipher
    cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend())
    decryptor = cipher.decryptor()

    # Decrypt the data
    decrypted_data = decryptor.update(encrypted_bytes) + decryptor.finalize()

    padding_length = decrypted_data[-1]
    if padding_length > 16:  # Invalid padding
        raise ValueError("Invalid padding detected.")
    decrypted_data = decrypted_data[:-padding_length]

    return decrypted_data.decode('utf-8')

if __name__ == "__main__":
    key = b"<webhook_key>"

    encrypted_data = "<encrypted_string>"

    try:
        decrypted_text = decrypt_aes_128_cbc(encrypted_data, key)
        print("Decrypted text:", decrypted_text)
    except Exception as e:
        print("Error during decryption:", e)
